TrailBlaze/View Controllers/RerunViewController.m

```diff
@@ -0,0 +1,92 @@
+//
+//  RerunViewController.m
+//  TrailBlaze
+//
+//  Created by Adam Issah on 8/3/22.
+//
+
+#import "RerunViewController.h"
+#import "Run.h"
+#import "RerunCell.h"
+#import "SceneDelegate.h"
+#import "HomeViewController.h"
+
+@interface RerunViewController () <UITableViewDataSource, UITableViewDelegate>
+@property (weak, nonatomic) IBOutlet UITableView *tableView;
+
+@end
+
+@implementation RerunViewController {
+    NSArray *pastRuns;
+}
+- (void) viewWillAppear:(BOOL)animated {
+    [Run retreiveRunObjects:PFUser.currentUser limit:10 completion:^(NSArray * _Nonnull runObjects, NSError * _Nullable err) {
+        if (runObjects) {
+            self->pastRuns = runObjects;
+            [self->_tableView reloadData];
+        } else {
+            NSLog(@"No past runs");
+        }
+    }];
+}
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    self.tableView.dataSource = self;
+    self.tableView.delegate = self;
+    [[PFUser currentUser][@"pastRuns"] fetchIfNeeded];
+    pastRuns = [PFUser currentUser][@"pastRuns"];
+    
+    PFFileObject *image = [PFUser.currentUser objectForKey:@"profileImage"];
+    NSLog(@"%@",image.url);
+    
+    [Run retreiveRunObjects:PFUser.currentUser limit:20 completion:^(NSArray * _Nonnull runObjects, NSError * _Nullable err) {
+        if (runObjects) {
+            self->pastRuns = runObjects;
+            [self->_tableView reloadData];
+        } else {
+            NSLog(@"No past runs");
+        }
+    }];
+    
+    
+}
+
+
+- (nonnull UITableViewCell *)tableView:(nonnull UITableView *)tableView cellForRowAtIndexPath:(nonnull NSIndexPath *)indexPath {
+    RerunCell *cell = [tableView dequeueReusableCellWithIdentifier:@"RerunCell" forIndexPath:indexPath];
+    if (pastRuns) {
+        @try {
+        cell.startLocationAddress.text = pastRuns[indexPath.row][@"startLocationAddress"];
+        cell.endLocationAddress.text = pastRuns[indexPath.row][@"endLocationAddress"];
+//        cell.averagePace.text = pastRuns[indexPath.row][@"overallAveragePace"];
+            cell.averagePace.text = @"9'32";
+        }
+        @catch (NSException * e) {
+            
+        }
```

Comment:
is the try/catch to guard against index out of bounds errors? better practice to check that explicitly. usually you only use try catch for very specific thrown exceptions and not for general ones.

Reply:
It is there to guard against instances where I did not end the run and closed the app so the run object is missing endTime, duration fields which causes the app to crash because those expected string will return as NSNull. In those cases I consider the run void and don't display it to the user as a completed run